src/webdav/handlers/PROPFIND.handler.ts

```diff
@@ -74,13 +78,44 @@ export class PROPFINDRequestHandler implements WebDavMethodHandler {
       encodeURI(resource.url),
     );
     const xml = XMLUtils.toWebDavXML([driveFile], {
-      arrayNodeName: 'response',
+      arrayNodeName: XMLUtils.addDefaultNamespace('response'),
     });
 
     return xml;
   }
 
-  private async getFolderContentXML(relativePath: string, folderUuid: string) {
+  private async getFolderContentXML(relativePath: string, folderUuid: string, depth: string, isRootFolder = false) {
+    let XMLNodes: object[] = [];
+
+    switch (depth) {
+      case '0':
+        if (isRootFolder) {
+          XMLNodes.push(await this.getFolderRootXMLNode(relativePath, folderUuid));
+        } else {
+          XMLNodes.push(await this.getFolderXMLNode(relativePath, folderUuid));
+        }
+        break;
+      case '1':
+      default:
+        if (isRootFolder) {
+          XMLNodes.push(await this.getFolderRootXMLNode(relativePath, folderUuid));
```

Comment:
I think we will need to move the XML generation logic to somewhere else at some point, this handler is starting to grow a lot, but not for now